`pytest-xdist` workers crashing

[Workflow Run URL](https://github.com/coiled/coiled-runtime/actions/runs/3154650491)

`tests/benchmarks/test_csv.py::test_csv_basic ` and `tests/benchmarks/test_dataframe.py::test_dataframe_align` caused `pytest-xdist` workers to crash for some reason. We've started seeing this with other tests on PRs as well. It's not clear why this is happening. It's not always these tests that cause `pytest` worker crashes -- other tests do as well. 

<details>

```python
[gw2] node down: Not properly terminated
[gw2] [  4%] FAILED tests/benchmarks/test_csv.py::test_csv_basic 

replacing crashed worker gw2
```

```python
[gw3] node down: Not properly terminated
[gw3] [  4%] FAILED tests/benchmarks/test_dataframe.py::test_dataframe_align 

replacing crashed worker gw3
```

</details>

We saw something similar in https://github.com/coiled/coiled-runtime/issues/412 with `tests/benchmarks/test_array.py::test_anom_mean`

```python
[gw0] node down: Not properly terminated
[gw0] [  4%] FAILED tests/benchmarks/test_array.py::test_anom_mean 

replacing crashed worker gw0
```

We saw something similar in https://github.com/coiled/coiled-runtime/issues/413 with `tests/benchmarks/test_zarr.py::test_select_scalar`


```python
[gw1] node down: Not properly terminated
[gw1] [  4%] FAILED tests/benchmarks/test_zarr.py::test_select_scalar 

replacing crashed worker gw1
```

--- a/.github/workflows/tests.yml
+++ b/.github/workflows/tests.yml
@@ -179,7 +179,7 @@ jobs:
   regressions:
     needs: process-results
     # Always check for regressions, as this can be skipped even if an indirect dependency fails (like a test run)
-    if: always()
+    if: always() && github.event_name != 'pull_request'
     name: Detect regressions
     runs-on: ubuntu-latest
     steps:

--- a/ci/scripts/run_tests.sh
+++ b/ci/scripts/run_tests.sh
@@ -16,4 +16,4 @@ then
   EXTRA_OPTIONS="$EXTRA_OPTIONS --benchmark"
 fi
 
-python -m pytest -n 8 --dist loadscope $EXTRA_OPTIONS $@
+python -m pytest -n 4 --dist loadscope $EXTRA_OPTIONS $@

--- a/dashboard.py
+++ b/dashboard.py
@@ -109,9 +109,9 @@ def calc_ab_confidence_intervals(
     # DataFrame with 10,000 rows per test exactly, with columns
     # [fullname, fullname_no_category, bootstrap_run, {A}, {B}, diff]
     pivot = bootstrapped.pivot(
-        ["fullname", "fullname_no_category", "bootstrap_run"],
-        "runtime",
-        field_name,
+        index=["fullname", "fullname_no_category", "bootstrap_run"],
+        columns="runtime",
+        values=field_name,
     ).reset_index()
     pivot["diff"] = pivot[B] / pivot[A] - 1
 
@@ -244,7 +244,10 @@ def make_ab_confidence_map(
     ]
     runtimes = df["runtime"].unique()
     A = baseline
-    B = next(r for r in runtimes if r != baseline)
+    B = next((r for r in runtimes if r != baseline), None)
+    if not B:
+        # Nothing to compare against
+        return None, 0
     conf = calc_ab_confidence_intervals(df, spec.field_name, A, B)
 
     n_bars = df["fullname_no_category"].unique().size

